packages/blade/src/components/BottomSheet/_decisions/decisions.md

```diff
@@ -0,0 +1,296 @@
+
+<details>
+
+<summary>Consumer side bottomsheet lazy loading example</summary>
+
+```jsx
+import { Spinner, useTheme, useBreakpoint } from '@razorpay/blade';
+
+const BottomSheet = React.lazy(() => import('@razorpay/blade/components/bottomsheet'));
+const DropdownOverlay = React.lazy(() => import('@razorpay/blade/components/dropdown-overlay'));
+
+const App = () => {
+  const { theme } = useTheme();
+  const { matchedDeviceType } = useBreakpoint({ breakpoints: theme.breakpoints });
+  const isMobile = matchedDeviceType === 'mobile';
+
+  return (
+    <Dropdown selectionType={selectionType}>
+      <SelectInput label="Select Action" />
+      {/* We can either put a fallback spinner or show skeleton loaders */}
+      <React.Suspense fallback={<Spinner />}>
+        {isMobile ? (
+          <BottomSheet>
+            <BottomSheetHeader />
+            <BottomSheetBody>
+              <SelectContent />
+            </BottomSheetBody>
+            <BottomSheetFooter />
+          </BottomSheet>
+        ) : (
+          <DropdownOverlay>
+            <SelectContent />
+          </DropdownOverlay>
+        )}
+      </React.Suspense>
+    </Dropdown>
+  );
+};
+
+const SelectContent = () => {
+  return (
+    <ActionList>
+      <ActionListItem
+        leading={<ActionListItemIcon icon={SettingsIcon} />}
+        title="Settings"
+        value="settings"
+      />
+      <ActionListItem leading={<ActionListItemIcon icon={InfoIcon} />} title="Info" value="info" />
+    </ActionList>
+  );
+};
+```
+
+</details>
+
+## Accessibility
+
+The bottom sheet will follow the accessibility characteristics of a Modal.
+
+[APG Guidelines for Modal](https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/)
+
+Behaviours:
+
+- Tab, Shift+Tab should cycle the focused elements
+- The focus should be trapped inside the bottom sheet
+- Escape closes the bottom sheet, after closing the focus will go to the triggered element
+- When opening the bottom sheet it should either focus on the first focusable element on the sheet or element provided by the user via `initialFocusRef` prop
+
+## Open questions
+
+- ~~What is `action` in the header trailing visual?~~
+- ~~In BottomSheetHeaderLeaing & BottomSheetFooterLeading will the `prefix` only support Icon component?~~
+
+- Consideration for using custom solution rather than third party libraries like [react-spring-bottomsheet](https://github.com/stipsan/react-spring-bottom-shee) in web.
+   - **ANS**: We did a initial research for various third-party libraries documented in [here](https://docs.google.com/document/d/1lRIBo5nkp1R9_DC9nRiFFwBEH3oVhMZjLbV7jxFrYBA/edit), most of them did not fit out use case, react-spring-bottomsheet had lot of features covered but comes with a bit of bundle size baggage like usegesture/xstate/xstate-react/react-spring libraries which in mobile users will have to carry.
```

Comment:
Move to a md file here please, I'm unable to access this doc.

Reply:
Done.